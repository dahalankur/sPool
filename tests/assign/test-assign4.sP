# lists can be assigned too 
myList = [1,2,3,4,5]
myList2 = myList

heterogeneous = [true, false, [1]] # parser does not prohibit heterogeneous lists

myList3 = getList(3, 1, true)
isTrue = false
myGpa = 2.0 # just an example to show floats can be assigned to variables

# this is a comment

# assigning string literals with non-ascii characters
myFunkyString = "this is a string with non-ascii characters: 你好€ÒÓ˝”"
