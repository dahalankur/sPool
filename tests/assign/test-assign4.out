PROGRAM {[ 
    ASSIGN(myList, LIST(INT(1), INT(2), INT(3), INT(4), INT(5))) 
    ASSIGN(myList2, VAR(myList)) 
    ASSIGN(heterogeneous, LIST(BOOL(true), BOOL(false), LIST(INT(1)))) 
    ASSIGN(myList3, CALL(getList, ARGS(INT(3), INT(1), BOOL(true)))) 
    ASSIGN(isTrue, BOOL(false)) 
    ASSIGN(myGpa, FLOAT(2.0)) 
    ASSIGN(myFunkyString, STRING("this is a string with non-ascii characters: 你好€ÒÓ˝”"))]}
